tests/MongoDB.Bson.TestHelpers/Reflector.cs

```diff
@@ -35,19 +35,33 @@ public static object GetPropertyValue(object obj, string name, BindingFlags flag
 
         public static object Invoke(object obj, string name, BindingFlags flags = BindingFlags.NonPublic | BindingFlags.Instance)
         {
-            var methodInfo = obj.GetType().GetMethods(flags)
-                .Where(m => m.Name == name && m.GetParameters().Length == 0)
-                .Single();
-            return methodInfo.Invoke(obj, new object[] { });
+            try
+            {
+                var methodInfo = obj.GetType().GetMethods(flags)
+                    .Where(m => m.Name == name && m.GetParameters().Length == 0)
+                    .Single();
+                return methodInfo.Invoke(obj, new object[] { });
+            }
+            catch (TargetInvocationException exception)
+            {
+                throw exception.InnerException;
+            }
```

Comment:
When testing that a method throws an exception we want the original exception thrown, not the TargetInvocationException that reflection wraps it in.